crates/relayer-types/src/core/ics04_channel/packet.rs

```diff
@@ -178,6 +181,32 @@ impl Packet {
 
         height_timed_out || timestamp_timed_out
     }
+
+    pub fn validate_fields(
+        &self,
+        max_memo_size: usize,
+        max_receiver_size: usize,
+    ) -> Result<(usize, usize), Error> {
+        let any_packet: RawPacketData =
+            serde_json::from_slice(&self.data).map_err(Error::serde_json_error)?;
```

Comment:
Here we should probably not return an error if the packet data is not an ICS-20 packet, but rather return `Ok((0, 0))` or something like that.